src/App.tsx

```diff
@@ -17,59 +23,137 @@ const goodsFromServer = [
+      case SortType.ALPHABET:
+        return g1.localeCompare(g2);
 
-  return visibleGoods;
+      case SortType.LENGTH:
+        return g1.length - g2.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  return isReversed
+    ? visibleGoods.reverse()
+    : visibleGoods;
 }
 
-// DON'T save goods to the state
 type State = {
   isStarted: boolean,
   isReversed: boolean,
   sortType: SortType,
 };
 
-export const App = () => (
-  <div className="App">
-    <button type="button">
-      Start
-    </button>
-
-    <button type="button">
-      Sort alphabetically
-    </button>
-
-    <button type="button">
-      Sort by length
-    </button>
-
-    <button type="button">
-      Reverse
-    </button>
-
-    <button type="button">
-      Reset
-    </button>
-
-    <ul className="Goods">
-      <li className="Goods__item">Dumplings</li>
-      <li className="Goods__item">Carrot</li>
-      <li className="Goods__item">Eggs</li>
-      <li className="Goods__item">...</li>
-    </ul>
-  </div>
-);
+export class App extends Component<{}, State> {
+  state = {
+    isStarted: false,
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  start = () => this.setState({ isStarted: true });
+
+  sortAlphabet = () => this.setState({ sortType: SortType.ALPHABET });
+
+  sortLength = () => this.setState({ sortType: SortType.LENGTH });
+
+  reverse = () => {
+    this.setState(state => ({ isReversed: !state.isReversed }));
+  };
+
+  reset = () => {
+    this.setState({
+      isReversed: false,
+      sortType: SortType.NONE,
+    });
+  };
+
+  render() {
+    const { isStarted, sortType, isReversed } = this.state;
+    const goods = getReorderedGoods(goodsFromServer, sortType, isReversed);
+
+    return (
+      <div className="App">
+
+        {isStarted
+          ? (
+            <>
+              <Button
+                type="button"
+                variant="outlined"
+                onClick={this.sortAlphabet}
+              >
+                Sort alphabetically
+              </Button>
+
+              <Button
+                type="button"
+                variant="outlined"
+                onClick={this.sortLength}
+              >
+                Sort by length
+              </Button>
+
+              <Button
+                type="button"
+                variant="outlined"
+                onClick={this.reverse}
+              >
+                Reverse
+              </Button>
+
+              <Button
+                type="button"
+                variant="outlined"
+                onClick={this.reset}
+              >
+                Reset
+              </Button>
+
+              <List
+                sx={{
+                  width: '100%',
+                  maxWidth: 360,
+                  bgcolor: 'background.paper',
+                }}
+                className="Goods"
+              >
+                {goods.map(good => (
+                  <ListItem
+                    key={good}
```

Comment:
Its better not to use potentially duplicate key. Watch the result of key duplicated in console => https://codesandbox.io/s/white-sound-lg2ntl?file=/src/App.js and try to remove item which have not unique key

You can use "uuid" library to create unique keys.

Reply:
I used it, but there's written in the checklist. 
![image](https://user-images.githubusercontent.com/33294729/183924025-315f4441-b511-4883-8aae-3d47744d3f4c.png)